From: Luca Weiss <luca@lucaweiss.eu>
Date: Mon, 11 Aug 2025 18:20:39 +0200
Subject: arm64: dts: qcom: qcm6490-shift-otter: Add 'Custom' button

On the left side of the phone is a button, marketed as 'Custom key'.
Describe it in the devicetree.

Signed-off-by: Luca Weiss <luca@lucaweiss.eu>
---
 arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts b/arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts
index c2c37fb4..51a6ed5 100644
--- a/arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts
+++ b/arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts
@@ -53,7 +53,7 @@ framebuffer0: framebuffer@a000000 {
 	gpio-keys {
 		compatible = "gpio-keys";
 
-		pinctrl-0 = <&volume_down_default>;
+		pinctrl-0 = <&volume_down_default>, <&gpio_custom_key_default>;
 		pinctrl-names = "default";
 
 		key-volume-up {
@@ -62,6 +62,13 @@ key-volume-up {
 			linux,code = <KEY_VOLUMEUP>;
 			debounce-interval = <15>;
 		};
+
+		key-custom {
+			label = "Custom Key";
+			gpios = <&tlmm 75 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_MISC>;
+			debounce-interval = <15>;
+		};
 	};
 
 	pmic-glink {
@@ -914,6 +921,13 @@ disp_reset_n_suspend: disp-reset-n-suspend-state {
 		bias-pull-down;
 	};
 
+	gpio_custom_key_default: gpio-custom-key-default-state {
+		pins = "gpio75";
+		function = "gpio";
+		drive_strength = <2>;
+		bias-pull-up;
+	};
+
 	mdp_vsync: mdp-vsync-state {
 		pins = "gpio80";
 		function = "mdp_vsync";
